### ISSUE TITLE
bug(forge coverage): lcov randomly doesn't generate - How to debug?

### ISSUE THREAD

**Issue by MarcusWentz** [foundry-rs/foundry#8841]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 

### What command(s) is the bug in?

```
forge coverage --report lcov
```

### Operating System

Linux

### Describe the bug

I was working on a Foundry project with multiple contracts for a few weeks.
I had no issues with compilation or test coverage during this time. 

However today I cannot seem to get the lcov file generation output from:
```
forge coverage --report lcov
```
to generate the HTML code report:
```
genhtml lcov.info -o report --branch-coverage
```
I tried adding more verbosity for coverage:
```
forge coverage --report lcov -vvvvv
```
but it just shows the EVM trace in detail without hints on why lcov file generation might be failing.
I also tried deleting and cloning the project multiple times but the issue is still there.
I also ran:
```
forge clean
```
and:
```
foundryup
```
but that didn't fix this either. 

The project is private but I can share some error logs.

Might be related to this with NATSPEC comments (Which UTF-8 characters cause lcov issues?):

https://github.com/foundry-rs/foundry/issues/8395#issuecomment-2218333994

?

What might be causing this? 

Is there a way to debug this?

**Comment by grandizzy** [comment]
@MarcusWentz pls double check there's no test failing, coverage would not generate if any test fail

**Comment by MarcusWentz** [comment]
I commented out multiple test contracts.
With just a basic contract that didn't require me to fork Base Sepolia or Base Mainnet (unit test), I ran:
```
forge coverage --report lcov
```
and the lcov file was generated. 

For the condition:
```
"coverage would not generate if any test fail"
```
was this added with this Foundry release yesterday:
https://github.com/foundry-rs/foundry/releases/tag/nightly-0079a1146b79a4aeda58b0258215bedb1f92700b
or was this always the case?
I recall being able to get lcov and HTML reports with some tests failing.

I am guessing I should just comment out tests on different networks when I switch between forking networks:

Base Sepolia:
```
forge coverage --fork-url $baseSepoliaHTTPS --report lcov && genhtml lcov.info -o report --branch-coverage
```
Base Mainnet:
```
forge coverage --fork-url $baseMainnetHTTPS --report lcov && genhtml lcov.info -o report --branch-coverage
```
?
